pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -186,14 +187,17 @@ func NewAttachDetachController(
 		adc.nodeStatusUpdater,
 		recorder)
 
+	adc.csiMigratedPluginManager = csiMigratedPluginManager
+
 	adc.desiredStateOfWorldPopulator = populator.NewDesiredStateOfWorldPopulator(
 		timerConfig.DesiredStateOfWorldPopulatorLoopSleepPeriod,
 		timerConfig.DesiredStateOfWorldPopulatorListPodsRetryDuration,
 		podInformer.Lister(),
 		adc.desiredStateOfWorld,
 		&adc.volumePluginMgr,
 		pvcInformer.Lister(),
-		pvInformer.Lister())
+		pvInformer.Lister(),
+		adc.csiMigratedPluginManager)
```

Comment:
Haven't looked at the code closely enough to determine whether this is feasabile (or wanted), but instead of having a second `PluginManager` here could we instead get it to implement the same plugin manager interface as the original `volumePluginMgr` then
```
if CSIMigrationInTreeOff{
// Inject csiMigratedPluginManager
} else{
// Inject volumePluginMgr
}
```

That way we don't choose between the two. If Migration has already been on and commited, the "migrated plugin manager" is just our standard one.

Edit: After reading some more it seems like the functionality of `MigratedPluginManager` is not the same as the original `volumePluginManager`. So maybe this isn't a good idea